### PR TITLE
Feature: add configuration test

### PR DESCRIPTION
**Pull request by whtsky**
I generate clash config with [clash-config-preprocessor](https://github.com/whtsky/clash-config-preprocessor/). Sometimes the generated config file is invalid due to upstream rule-set changes, so it would be nice to have the configuration file test feature built-in.

The cli design is highly influenced by `nginx -t`.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -57,6 +61,19 @@ func main() {
 		log.Fatalln("Initial configuration directory error: %s", err.Error())
 	}
 
+	if testConfig {
+		// silence INFO logs during config parsing
+		log.SetLevel(log.WARNING)
```
Comment (Dreamacro) [inline main.go]: why not display provider log

Comment (whtsky) [reply main.go]: IMO display `Start initial compatible provider XXX` during configuration test is confusing.

https://github.com/Dreamacro/clash/blob/609869bf5a1d779b0ac6a0caca9ff9ea67b0c40f/config/config.go#L331

Comment (Dreamacro) [reply main.go]: but it does a part of the parse.

Comment (whtsky) [reply main.go]: 图片

Comment (whtsky) [reply main.go]: removed

File: main.go [hunk 2]
```diff
@@ -57,6 +61,17 @@ func main() {
 		log.Fatalln("Initial configuration directory error: %s", err.Error())
 	}
 
+	if testConfig {
+		if _, err := executor.Parse(); err != nil {
+			fmt.Fprintln(os.Stderr, err.Error())
```
Comment (Dreamacro) [inline main.go]: I prefer all logs use `"github.com/Dreamacro/clash/log"`

if you want to split `warning` `error` into stderr, you can refer https://github.com/sirupsen/logrus/issues/403 and modify `"github.com/Dreamacro/clash/log"`

Comment (whtsky) [reply main.go]: Ok, I changed the error output to use log.

As for `configuration file %s test failed | success`, I believe they are not part of log but rather standard outputs - like `clash -v` and what nginx does:
图片

Comment (Dreamacro) [reply main.go]: fine

File: main.go [hunk 3]
```diff
@@ -57,6 +61,17 @@ func main() {
 		log.Fatalln("Initial configuration directory error: %s", err.Error())
 	}
 
+	if testConfig {
+		if _, err := executor.Parse(); err != nil {
+			log.Errorln(err.Error())
+			fmt.Printf("configuration file %s test failed\n", constant.Path.Config())
+			os.Exit(1)
+		} else {
```
Comment (Dreamacro) [inline main.go]: The last problem: the `else` is not necessary

**Comment by Dreamacro** [comment]
LGTM, Thanks 👍

**Pull request merged**